src/Interpreters/Cluster.cpp

```diff
@@ -268,18 +283,25 @@ Clusters::Impl Clusters::getContainer() const
 
 /// Implementation of `Cluster` class
 
-Cluster::Cluster(const Poco::Util::AbstractConfiguration & config, const Settings & settings, const String & cluster_name)
+Cluster::Cluster(const Poco::Util::AbstractConfiguration & config,
+                 const Settings & settings,
+                 const String & config_prefix_,
+                 const String & cluster_name)
 {
+    auto config_prefix = config_prefix_ + "." + cluster_name;
+
     Poco::Util::AbstractConfiguration::Keys config_keys;
-    config.keys(cluster_name, config_keys);
+    config.keys(config_prefix, config_keys);
 
-    if (config_keys.empty())
-        throw Exception("No cluster elements (shard, node) specified in config at path " + cluster_name, ErrorCodes::SHARD_HAS_NO_CONNECTIONS);
+    config_prefix += ".";
 
-    const auto & config_prefix = cluster_name + ".";
+    secret = config.getString(config_prefix + "secret", "");
+    std::erase(config_keys, "secret");
```

Comment:
Yandex synchronization check says
```
src/Interpreters/Cluster.cpp:299:10: error: no member named 'erase' in namespace 'std'
    std::erase(config_keys, "secret");
    ~~~~~^
1 error generated.
```

Please replace `std::erase` with `boost::range::remove_erase`.

Reply:
@vitlibar it uses unbundled libstdc++/libc++?

Reply:
That's quite an unusual build which links ClickHouse with a lot of closed source code and which still can't use most of C++20 features and also.